Google.PowerShell/BigQuery/GcbqTable.cs

```diff
@@ -112,4 +112,145 @@ protected override void ProcessRecord()
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByValues)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Dataset), Property = nameof(Apis.Bigquery.v2.Data.Dataset.DatasetReference))]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(DatasetReference), Property = nameof(DatasetReference.DatasetId))]
+        public string Dataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The TableId must be unique within the Dataset and match the pattern [a-zA-Z0-9_]+.
+        /// </para>
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = ParameterSetNames.ByValues)]
+        [ValidatePattern("[a-zA-Z0-9_]")]
+        [ValidateLength(1, 1024)]
+        public string TableId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// User-friendly name for the table.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Description of the table.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The lifetime of this table from the time of creation (in seconds).
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        [ValidateRange(3600, (long.MaxValue / 1000))]
+        public long Expiration { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            // Set up the Dataset based on parameters
+            Table newTab;
+            TablesResource.InsertRequest request;
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByObject:
+                    newTab = InputObject;
+                    request = Service.Tables.Insert(newTab, newTab.TableReference.ProjectId, 
+                        newTab.TableReference.DatasetId);
+                    break;
+                case ParameterSetNames.ByValues:
+                    newTab = new Table();
+                    newTab.TableReference = new TableReference();
+                    newTab.TableReference.DatasetId = Dataset;
+                    newTab.TableReference.ProjectId = Project;
+                    newTab.TableReference.TableId = TableId;
+                    newTab.FriendlyName = Name;
+                    newTab.Description = Description;
+                    if (Expiration != 0)
+                    {
+                        double currentMillis = DateTime.Now.ToUniversalTime().Subtract(
```

Comment:
You can use this built-in [method](https://msdn.microsoft.com/en-us/library/system.datetimeoffset.tounixtimemilliseconds(v=vs.110).aspx) to get milliseconds since epoch.